service/service.go

```diff
@@ -2,90 +2,161 @@ package service
+type DatsetAPIStore struct {
+	store.MongoDB
+	//	store.GraphDB
+}
+
 // Service contains all the configs, server and clients to run the dp-topic-api API
 type Service struct {
-	Config      *config.Config
-	Server      HTTPServer
-	Router      *mux.Router
-	API         *api.API
-	ServiceList *ExternalServiceList
-	HealthCheck HealthChecker
-	MongoDB     api.MongoServer
+	Config         *config.Configuration
+	ServiceList    *ExternalServiceList
+	Server         HTTPServer
+	Router         *mux.Router
+	API            *api.API
+	HealthCheck    HealthChecker
+	mongoDB        store.MongoDB
+	IdentityClient *clientsidentity.Client
 }
 
-// Run the service
-func Run(ctx context.Context, cfg *config.Config, serviceList *ExternalServiceList, buildTime, gitCommit, version string, svcErrors chan error) (*Service, error) {
-
-	log.Event(ctx, "running service", log.INFO)
-
-	log.Event(ctx, "using service configuration", log.Data{"config": cfg}, log.INFO)
-
-	// Get HTTP Server
-	router := mux.NewRouter()
-
-	// and ... // ADD CODE: Add any middleware that your service requires
-
-	srv := serviceList.GetHTTPServer(cfg.BindAddr, router)
+// New creates a new service
+func New(cfg *config.Configuration, serviceList *ExternalServiceList) *Service {
+	svc := &Service{
+		Config:      cfg,
+		ServiceList: serviceList,
+	}
+	return svc
+}
 
-	// ADD CODE: Add other(s) to serviceList here
+// Run the service
+func (svc *Service) Run(ctx context.Context, buildTime, gitCommit, version string, svcErrors chan error) (err error) {
 
 	// Get MongoDB client
-	mongoDB, err := serviceList.GetMongoDB(ctx, cfg)
+	svc.mongoDB, err = svc.ServiceList.GetMongoDB(ctx, svc.Config)
 	if err != nil {
 		log.Event(ctx, "failed to initialise mongo DB", log.FATAL, log.Error(err))
-		return nil, err
+		return err
 	}
+	store := store.DataStore{Backend: DatsetAPIStore{svc.mongoDB}}
 
-	// Setup the API
-	a := api.Setup(ctx, cfg, router, mongoDB)
-
-	var zebedeeClient *health.Client
-	if cfg.EnablePrivateEndpoints {
+	// Get Identity Client (only if private endpoints are enabled)
+	if svc.Config.EnablePrivateEndpoints {
 		// Only in Publishing ... create client(s):
-
-		zebedeeClient = serviceList.GetHealthClient("Zebedee", cfg.ZebedeeURL)
+		svc.IdentityClient = clientsidentity.New(svc.Config.ZebedeeURL)
 	}
 
-	hc, err := serviceList.GetHealthCheck(cfg, buildTime, gitCommit, version)
-
+	// Get HealthCheck
+	svc.HealthCheck, err = svc.ServiceList.GetHealthCheck(svc.Config, buildTime, gitCommit, version)
 	if err != nil {
 		log.Event(ctx, "could not instantiate healthcheck", log.FATAL, log.Error(err))
-		return nil, err
+		return err
 	}
 
-	if err := registerCheckers(ctx, cfg, hc, mongoDB, zebedeeClient); err != nil {
-		return nil, errors.Wrap(err, "unable to register checkers")
+	if err := svc.registerCheckers(ctx /*svc.Config, healthCheck, mongoDB, zebedeeClient*/); err != nil {
```

Comment:
I see that in registerCheckers the clients stored in the svc struct are used, so we will not need to pass any other parameter at this point. I would remove the comment `/*svc.Config, healthCheck, mongoDB, zebedeeClient*/`

Reply:
Removed.